qiskit_experiments/framework/cache_method.py

```diff
@@ -0,0 +1,144 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Method decorator for caching regular methods in class instances.
+"""
+
+from typing import Union, Dict, Callable
+import functools
+
+
+def cache_method(
+    cache: Union[Dict, str] = "_cache", cache_args: bool = True, require_hashable: bool = True
```

Comment:
Personally I find the booleans easier to understand than string values. I don't think we should add more flexibility in hashing mechanism. If anything maybe we should remove the `require_hashable` should be removed so the option is just to use all args (and they must be hashable) or none.

Reply:
If you remove that option perhaps this is only applicable to static methods? Because `self` of an experiment instance is not hashable.

Reply:
`self` not being hashed is the design of this decorator, and main reason why you need it instead of `lru_cache`. So it should only be used with regular methods, not static methods. For static methods you should be able to use a regular `lru_cache` without issues since it behaves like a regular function.

Reply:
PS @nkanazawa1989 I will make a commit to remove the require_hashable kwarg to not overcomplicate this as you suggest.